data/ai/micro_ais/micro_ai_helper.lua

```diff
@@ -164,32 +214,42 @@ function micro_ai_helper.micro_ai_setup(cfg, CA_parms, required_keys, optional_k
     local CA_cfg = {}
 
     -- Required keys
-    for _,v in pairs(required_keys) do
-        if v:match('%[[a-zA-Z0-9_]+%]')  then
-            v = v:sub(2,-2)
-            if not wml.get_child(cfg, v) then
-                wml.error("[micro_ai] tag (" .. cfg.ai_type .. ") is missing required parameter: [" .. v .. "]")
+    for k_name,k_type in pairs(required_keys) do
+        if (k_type == 'tag') then
+            -- Check that this is not a scalar parameter
+            if cfg[k_name] then
```

Comment:
This is probably fine in the vast majority of cases, but keep in mind that it's possible to have a tag and a key with the same name. I think in that case it makes more sense to issue a warning rather than an error, if that's not too much work.

I guess it's not a priority though.